climada/engine/test/test_forecast.py

```diff
@@ -0,0 +1,227 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Test Forecast class
+"""
+
+import unittest
+import datetime as dt
+import numpy as np
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import fiona
+from cartopy.io import shapereader
+
+from climada import CONFIG
+from climada.hazard.storm_europe import StormEurope
+from climada.entity.exposures.base import Exposures, INDICATOR_IF
+from climada.entity import ImpactFuncSet
+from climada.entity.impact_funcs.storm_europe import IFStormEurope
+from climada.engine.forecast import Forecast
+from climada.util.constants import WS_DEMO_NC
+
+HAZ_DIR = CONFIG.hazard.test_data.dir()
+
+class TestCalc(unittest.TestCase):
+    """Test calc and propety functions from the Forecast class"""
+
+    def test_Forecast_calc_properties(self):
+        """Test calc and propety functions from the Forecast class"""
+        #hazard
+        haz = StormEurope()
+        haz.read_cosmoe_file(HAZ_DIR.joinpath('storm_europe_cosmoe_forecast_vmax_testfile.nc'),
+                             run_datetime=dt.datetime(2018,1,1),
+                             event_date=dt.datetime(2018,1,3))
+        #exposure
+        data = {}
+        data['latitude'] = haz.centroids.lat
+        data['longitude'] = haz.centroids.lon
+        data['value'] = np.ones_like(data['latitude']) * 100000
+        data['deductible'] = np.zeros_like(data['latitude'])
+        data[INDICATOR_IF + 'WS'] = np.ones_like(data['latitude'])
+        # data['category_id'] = np.array([1, 2, 3])
+        data['region_id'] = np.ones_like(data['latitude'],dtype=int) * 756
+        expo = Exposures(gpd.GeoDataFrame(data=data))
+        #vulnerability
+        #generate vulnerability
+        impact_function = IFStormEurope()
+        impact_function.set_welker()
+        impact_function_set = ImpactFuncSet()
+        impact_function_set.append(impact_function)
+        #create and calculate Forecast
+        forecast = Forecast({dt.datetime(2018,1,1): haz}, expo, impact_function_set)
```

Comment:
A test with a dict with more than one hazard would be good.

Reply:
I tried to sneak that into the plot test at line 142. Does this suffice or should I write a specific test function for this? (Each new test function takes about 1.5 seconds, because creating hazard, exposures, impactfunctionset and do the calc() is expensive...)

Reply:
A ok sounds good then!